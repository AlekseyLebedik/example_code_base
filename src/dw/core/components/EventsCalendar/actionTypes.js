export const SET_EVENTS_CALENDAR_SETTINGS = 'SET_EVENTS_CALENDAR_SETTINGS';
export const CLEAR_EVENTS_CALENDAR_SETTINGS = 'CLEAR_EVENTS_CALENDAR_SETTINGS';
