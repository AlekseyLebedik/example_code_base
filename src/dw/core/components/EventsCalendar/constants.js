export const LOCAL_STORAGE_CALENDAR_SETTINGS = 'calendarSettings';
export const LOCAL_STORAGE_PRESETS = 'calendarPreset';
export const CALENDAR_PRESETS_SETTINGS = 'calendar_presets_settings';

export const IGNORE_PRESET_KEYS = [
  'calendarSettingsVersion',
  'datePickerProps',
  'eventTimeOffset',
  'groupLoadingStatuses',
  'selectedDay',
  'sidebarHovering',
];

export const FILTER_TABS = ['DEFAULT', 'PLATFORMS', 'CUSTOM'];

export const ENV_TYPE_FILTERS = {
  Certification: 'cert',
  Development: 'dev',
  Live: 'live',
};

export const ENV_TYPES = ['live', 'dev', 'cert'];

export const EXTERNAL_EVENTS_TYPES = ['holidays', 'pmg'];

export const PLATFORM_FILTERS = [
  'CROSSPLAY',
  'PS5',
  'PS4',
  'XB1',
  'XBSX',
  'PSN',
  'XBL',
  'BNET',
  'PC',
  'STEAM',
  'SWITCH',
  'Multiple',
  'Unspecified',
];

export const DEMONWARE_EVENTS = [
  { id: 'criticalEvents', title: 'Critical Events' },
  { id: 'maintenance', title: 'Maintenance' },
  { id: 'incidents', title: 'Incidents' },
  { id: 'generalComments', title: 'General Comments' },
];

export const ABTESTING_STATUSES = [
  'active',
  'analysis',
  'archived',
  'config',
  'killed',
];

export const EXPYTESTS_STATUSES = ['proposed', 'approved'];

export const STORY_FILTERS = ['Group', 'Timewarp', 'None'];

export const EVENT_GROUPS_SCHEMA = {
  properties: {
    CustomEvent: { type: 'object' }, // Component
    CustomEventProps: {
      type: 'object',
      properties: { displayTimeWithTitle: { type: 'boolean' } },
    },
    GroupLink: { type: 'object' }, // Component
    classes: {
      type: 'object',
      patternProperties: {
        // Style class pattern is <group_type>-<event_type>(-<event_status>)?
        '^(w)+-(w)+-*(w)*$': { type: 'string' },
      },
    },
    customEventStyles: { type: 'object' }, // function
    eventDragDrop: { type: 'object' }, // function
    eventTypes: {
      type: 'object',
      properties: {
        name: { type: 'string' },
        children: {
          type: 'array',
          items: { types: 'object', properties: { name: { type: 'string' } } },
        },
        selectedByDefault: { type: 'boolean' },
      },
    },
    events: {
      type: 'array',
      items: {
        type: 'object',
        properties: {
          end_at: { type: 'number' },
          env_type: {
            type: 'string',
            enum: ['Live', 'Certification', 'Development', 'Cross environment'],
          },
          event_type: {
            type: 'string',
            enum: [
              'eventManager',
              'demonwareEvents',
              'informationalEvents',
              'externalEvents',
              'abTesting',
              'expyTests',
            ],
          },
          id: { type: 'number' },
          is_deleted: { type: 'boolean' },
          is_manually_locked: { type: 'boolean' },
          is_private: { type: 'boolean' },
          is_restricted: { type: 'boolean' },
          is_schedule: { type: 'boolean' },
          is_template: { type: 'boolean' },
          manual_tags: {
            type: 'array',
            items: {
              type: 'string',
            },
          },
          platforms: {
            type: 'array',
            items: {
              type: 'string',
              enum: PLATFORM_FILTERS,
            },
          },
          project: { type: 'number' },
          projects: {
            type: 'array',
            items: {
              type: 'number',
            },
          },
          publish_at: { type: 'number' },
          status: {
            type: 'array',
            items: {
              type: 'string',
              enum: [
                'active',
                'approved',
                'cancelled',
                'ended',
                'eventManager',
                'expired',
                'open',
                'pending',
                'published',
                'rejected',
                'scheduled',
                'status',
              ],
            },
          },
          title: { type: 'string' },
        },
      },
    },
    loading: {
      type: 'object',
      properties: {
        error: { type: 'boolean' },
        isLoading: { type: 'boolean' },
      },
    },
    modalId: { type: 'string' },
    modifyEvent: { type: 'object' }, // function
    onSelectEvent: { type: 'object' }, // function
    type: {
      type: 'string',
      enum: [
        'eventManager',
        'demonwareEvents',
        'informationalEvents',
        'externalEvents',
        'abTesting',
      ],
    },
    wrapper: { type: 'object' }, // function
  },
};

export const CALENDAR_SETTINGS_SCHEMA = {
  properties: {
    customViewOn: { type: 'boolean' },
    displayView: {
      type: 'string',
      enum: ['calendar', 'list', 'timeline'],
    },
    filters: {
      type: 'object',
      properties: {
        customTags: {
          type: 'object',
          required: ['unspecified', 'userTags'],
          properties: {
            unspecified: {
              type: 'boolean',
              default: false,
            },
            userTags: {
              type: 'object',
              patternProperties: {
                '^.+$': {
                  type: 'object',
                  required: ['checked', 'tags'],
                  properties: {
                    checked: {
                      type: 'boolean',
                      default: false,
                    },
                    tags: {
                      type: 'array',
                      items: {
                        type: 'string',
                      },
                    },
                  },
                },
              },
            },
          },
        },
        environments: {
          type: 'object',
          required: [
            'Development',
            'Certification',
            'Live',
            'Cross Environment',
          ],
          properties: {
            Development: {
              type: 'boolean',
              default: false,
            },
            Certification: {
              type: 'boolean',
              default: false,
            },
            Live: {
              type: 'boolean',
              default: false,
            },
            'Cross Environment': {
              type: 'boolean',
              default: false,
            },
          },
        },
        gamertags: {
          type: 'object',
          patternProperties: {
            '^.+$': {
              type: 'boolean',
            },
          },
        },
        platforms: {
          type: 'object',
          patternProperties: {
            '^.+$': {
              type: 'boolean',
            },
          },
        },
        projects: {
          type: 'object',
          patternProperties: {
            '^.+$': {
              type: 'boolean',
            },
          },
        },
        sources: {
          type: 'object',
          properties: {
            abTesting: {
              type: 'object',
              title: 'The Abtesting Schema',
              required: ABTESTING_STATUSES,
              properties: {
                active: {
                  type: 'boolean',
                  default: false,
                },
                analysis: {
                  type: 'boolean',
                  default: false,
                },
                archived: {
                  type: 'boolean',
                  default: false,
                },
                config: {
                  type: 'boolean',
                  default: false,
                },
                killed: {
                  type: 'boolean',
                  default: false,
                },
              },
            },
            expyTests: {
              type: 'object',
              title: 'The EXPY Tests Schema',
              required: EXPYTESTS_STATUSES,
              properties: {
                approved: {
                  type: 'boolean',
                  default: false,
                },
                proposed: {
                  type: 'boolean',
                  default: false,
                },
              },
            },
            eventManager: {
              type: 'object',
              required: [
                'scheduled',
                'cancelled',
                'expired',
                'rejected',
                'ended',
                'published',
                'active',
                'open',
                'approved',
                'pending',
              ],
              properties: {
                scheduled: {
                  type: 'boolean',
                  default: false,
                },
                cancelled: {
                  type: 'boolean',
                  default: false,
                },
                expired: {
                  type: 'boolean',
                  default: false,
                },
                rejected: {
                  type: 'boolean',
                  default: false,
                },
                ended: {
                  type: 'boolean',
                  default: false,
                },
                published: {
                  type: 'boolean',
                  default: false,
                },
                active: {
                  type: 'boolean',
                  default: false,
                },
                open: {
                  type: 'boolean',
                  default: false,
                },
                approved: {
                  type: 'boolean',
                  default: false,
                },
                pending: {
                  type: 'boolean',
                  default: false,
                },
              },
            },
            informationalEvents: {
              type: 'object',
              properties: {
                demonware: {
                  type: 'boolean',
                  default: false,
                },
                'first-party': {
                  type: 'boolean',
                  default: false,
                },
                cdl: {
                  type: 'boolean',
                  default: false,
                },
              },
            },
            demonwareEvents: {
              type: 'object',
              properties: {
                criticalEvents: {
                  type: 'boolean',
                  default: true,
                },
                generalComments: {
                  type: 'boolean',
                  default: false,
                },
                incidents: {
                  type: 'boolean',
                  default: false,
                },
                maintenance: {
                  type: 'boolean',
                  default: false,
                },
              },
            },
            externalEvents: {
              type: 'object',
              properties: {
                holidays: {
                  type: 'object',
                  patternProperties: {
                    '^.+$': {
                      type: 'boolean',
                    },
                  },
                },
                'video-games': {
                  type: 'boolean',
                  default: false,
                },
                sports: {
                  type: 'boolean',
                  default: false,
                },
                pmg: {
                  type: 'object',
                  patternProperties: {
                    '^.+$': {
                      type: 'boolean',
                    },
                  },
                },
              },
            },
          },
        },
        stories: {
          type: 'object',
          required: ['Group', 'Timewarp', 'None'],
          properties: {
            Group: {
              type: 'boolean',
              default: true,
            },
            Timewarp: {
              type: 'boolean',
              default: true,
            },
            None: {
              type: 'boolean',
              default: true,
            },
          },
        },
      },
    },
    numberOfDays: { type: 'integer' },
    selectedView: {
      type: 'string',
      enum: ['day', 'week', 'month'],
    },
  },
};

export const STATUS_GROUPS = {
  active: 'Active',
  analysis: 'Analysis',
  approved: 'Approved',
  archived: 'Archived',
  cancelled: 'Cancelled',
  config: 'Config',
  ended: 'Ended',
  expired: 'Expired',
  killed: 'Killed',
  open: 'Open',
  pending: 'Pending',
  proposed: 'Proposed',
  published: 'Published',
  rejected: 'Rejected',
  scheduled: 'Scheduled',
};

export const HOUR = 3600;
export const DAY = 1000 * 60 * 60 * 24;
export const WEEK = DAY * 7;
export const MILLISECONDS_FORMAT = 1000;
