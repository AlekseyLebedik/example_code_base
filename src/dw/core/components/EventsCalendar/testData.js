import React from 'react';

import moment from 'moment-timezone';
import range from 'lodash/range';

import { dateToUTCTimestamp } from 'dw/core/helpers/date-time';

import { setViewComponents } from './helpers';

const eventsCalendarSettings = {
  customViewOn: false,
  envTypeFilters: {
    Development: true,
    Certification: true,
    Live: true,
  },
  eventTimeOffset: 0,
  filters: {
    customTags: {
      unspecified: true,
      userTags: {
        0: {
          checked: true,
          tags: ['PS4', 'XBOX'],
        },
        1: {
          checked: true,
          tags: ['COD', 'BO4'],
        },
      },
    },
    environments: {
      Development: true,
      Certification: true,
      Live: true,
      'Cross Environment': true,
    },
    gamertags: {},
    platforms: {
      CROSSPLAY: true,
      PC: true,
      PS4: true,
      SWITCH: true,
      XB1: true,
      Unspecified: true,
    },
    projects: {
      'GTR Project': true,
      'Call of Duty: Modern Warfare 2': true,
      'Permissions: Test 1': true,
      'Cross Project': true,
    },
    sources: {
      eventManager: {
        applied: false,
        approved: true,
        cancelled: false,
        expired: false,
        pending: false,
        rejected: false,
        scheduled: false,
        testing: false,
        'test-failed': false,
      },
      abTesting: {
        active: false,
        analysis: false,
        archived: false,
        config: false,
        killed: false,
      },
      informationalEvents: { demonware: true },
      demonwareEvents: {
        criticalEvents: true,
        generalComments: true,
        incidents: true,
        maintenance: true,
      },
      externalEvents: {
        holidays: true,
        pmg: { Other: true },
        'video-games': false,
        sports: false,
      },
    },
    stories: undefined,
  },
  displayView: 'calendar',
  groupLoadingStatuses: {
    eventManager: {
      error: null,
      isLoading: false,
    },
    abTesting: {
      error: null,
      isLoading: false,
    },
    informationalEvents: {
      error: null,
      isLoading: false,
    },
    demonwareEvents: {
      error: null,
      isLoading: false,
    },
    externalEvents: {
      error: null,
      isLoading: false,
    },
  },
  numberOfDays: 30,
  selectedDay: moment('2019-01-01T00:00:00Z'),
  selectedStyle: 'sources',
  selectedView: 'month',
};

const eventGroups = [
  {
    classes: {
      applied: 'applied-style',
      approved: 'approved-style',
      cancelled: 'cancelled-style',
      expired: 'expired-style',
      pending: 'pending-style',
      rejected: 'rejected-style',
      scheduled: 'scheduled-style',
      testing: 'testing-style',
      'test-failed': 'test-failed-style',
    },
    customEventStyles: event => event.status,
    EventDetails: () => <div />,
    events: [
      {
        allDay: false,
        id: 1,
        end: 1555460772,
        publish_at: 1555460772,
        start: 1555460772,
        status: 'approved',
        title: 'test',
        type: 'eventManager',
      },
      {
        allDay: false,
        id: 1,
        end: 1555460772,
        publish_at: 1555460772,
        start: 1555460772,
        status: 'pending',
        title: 'test 2',
        type: 'eventManager',
      },
    ],
    eventDragDrop: jest.fn(),
    eventTypes: {
      eventManager: {
        name: 'Event Manager',
        children: {
          checkedByMe: {
            name: 'Checked By Me',
            children: null,
            selectedByDefault: false,
          },
          status: {
            name: 'Status',
            children: {
              applied: {
                name: 'Applied',
                children: null,
                selectedByDefault: true,
              },
              approved: {
                name: 'Approved',
                children: null,
                selectedByDefault: true,
              },
              cancelled: {
                name: 'Cancelled',
                children: null,
                selectedByDefault: true,
              },
              expired: {
                name: 'Expired',
                children: null,
                selectedByDefault: true,
              },
              pending: {
                name: 'Pending',
                children: null,
                selectedByDefault: true,
              },
              rejected: {
                name: 'Rejected',
                children: null,
                selectedByDefault: true,
              },
              scheduled: {
                name: 'Scheduled',
                children: null,
                selectedByDefault: true,
              },
              testing: {
                name: 'Testing',
                children: null,
                selectedByDefault: true,
              },
              'test-failed': {
                name: 'Test Failed',
                children: null,
                selectedByDefault: true,
              },
            },
          },
        },
        selectedByDefault: true,
      },
    },
    GroupLink: () => <div />,
    loading: {
      error: null,
      isLoading: false,
    },
    modifyEvent: jest.fn(),
    type: 'eventManager',
    wrapper: jest.fn(),
  },
];

const filteredEvents = [
  {
    allDay: false,
    id: 1,
    end: 1555460772,
    publish_at: 1555460772,
    start: 1555460772,
    status: 'approved',
    title: 'test',
    type: 'eventManager',
  },
];

export const eventCalendarTestProps = {
  addCustomTagSet: jest.fn(),
  changeCalendarView: jest.fn(),
  changeNumberOfDays: jest.fn(),
  classes: {},
  clearEventsCalendarSettings: jest.fn(),
  copyLinkToClipboard: jest.fn(),
  customRangeError: '',
  customRangeInput: 30,
  customViewOn: false,
  createEventModalId: 'CREATE_EVENT',
  date: moment('2019-01-01T00:00:00Z').toDate(),
  daysWithDots: {},
  defaultDate: moment('2019-01-01T00:00:00Z'),
  disabledFilters: {},
  eventGroups,
  eventsCalendarSettings,
  eventTimeOffset: 0,
  filteredEvents,
  filteredGamertagGroups: [],
  handleSelectEvent: jest.fn(),
  handleOpenCreateEventModal: jest.fn(),
  history: { replace: jest.fn() },
  localizer: { format: jest.fn() },
  location: { search: '' },
  navigateCalendar: jest.fn(),
  onChangeDatePickerSelection: jest.fn(),
  onDrillDown: jest.fn(),
  onEventDrop: jest.fn(),
  onFetchEvents: jest.fn(),
  onNavigate: jest.fn(),
  onPresetsUpdate: jest.fn(),
  onSelectSlot: jest.fn(),
  onSetCalendarSettings: jest.fn(),
  permissions: {
    adminPermission: true,
    adminUser: true,
    emUser: true,
    eventWritePermission: true,
    staffUser: true,
  },
  platforms: ['CROSSPLAY', 'PC', 'PS4', 'SWITCH', 'XB1'],
  presetOptions: [],
  currentProject: { id: 1 },
  projectId: '1',
  setCalendarSettings: jest.fn(),
  setCustomEvent: jest.fn(),
  setDayMonthStyle: jest.fn(),
  setEventStyle: jest.fn(),
  setNonCustomDays: jest.fn(),
  showAllColors: false,
  sidebar: false,
  toastMsg: jest.fn(),
  toggleDisplayView: jest.fn(),
  userTimezone: 'UTC',
  views: setViewComponents(['day', 'week', 'month']),
};

const events = [
  {
    created_at: dateToUTCTimestamp(new Date(2019, 0, 1)),
    end: moment('2019-01-01T00:00:00Z').toDate(),
    publish_at: dateToUTCTimestamp(new Date(2019, 0, 1)),
    start: moment('2019-01-01T00:00:00Z').toDate(),
    status: 'pending',
    title: 'Test 1',
    type: 'eventManager',
  },
  {
    created_at: dateToUTCTimestamp(new Date(2019, 0, 1)),
    end: moment('2019-01-01T00:00:00Z').toDate(),
    publish_at: dateToUTCTimestamp(new Date(2019, 0, 1)),
    start: moment('2019-01-01T00:00:00Z').toDate(),
    status: 'approved',
    title: 'Test 2',
    type: 'eventManager',
  },
  {
    created_at: dateToUTCTimestamp(new Date(2019, 0, 1)),
    end: moment('2019-01-01T00:00:00Z').toDate(),
    publish_at: dateToUTCTimestamp(new Date(2019, 0, 1)),
    start: moment('2019-01-01T00:00:00Z').toDate(),
    status: 'pending',
    title: 'Test 3',
    type: 'eventManager',
  },
  {
    created_at: dateToUTCTimestamp(new Date(2019, 0, 8)),
    end: moment('2019-01-08T00:00:00Z').toDate(),
    publish_at: dateToUTCTimestamp(new Date(2019, 0, 8)),
    start: moment('2019-01-08T00:00:00Z').toDate(),
    status: 'pending',
    title: 'Test 4',
    type: 'eventManager',
  },
];

const timeGridProps = {
  accessors: {},
  components: {},
  getDrilldownView: jest.fn(),
  getNow: jest.fn(),
  getters: {},
};

export const eventCalendarChildrenTestProps = {
  ...eventCalendarTestProps,
  ...eventsCalendarSettings,
  ...timeGridProps,
  customFiltersSelected: false,
  events,
};

export const customMonthTestProps = {
  ...eventCalendarChildrenTestProps,
  closePopup: jest.fn(),
  filterWeekEvents: jest.fn(),
  getOverlayPlacement: jest.fn(),
  handleSelectSlot: jest.fn(),
  handleShowMore: jest.fn(),
  needLimitMeasure: false,
  maxRows: 7,
  monthRange: (customViewOn, numberOfDays, selectedDay) => {
    const start = customViewOn
      ? moment(selectedDay).startOf('week').subtract(1, 'day')
      : moment(selectedDay).startOf('month').startOf('week').subtract(1, 'day');
    const end = customViewOn
      ? moment(selectedDay)
          .add(numberOfDays - 1, 'day')
          .endOf('week')
      : moment(selectedDay).endOf('month').endOf('week');
    const numCalendarDays = end.diff(start, 'days');
    return range(0, numCalendarDays).map(() => start.add(1, 'day').toDate());
  },
  monthRef: {},
  onSelectEvent: jest.fn(),
  popup: true,
  popupEvents: [],
  position: {},
  renderHeader: jest.fn(),
  setDraggingHeaderGroup: jest.fn(),
  setPosition: jest.fn(),
};

export const sidebarDefaultTabProps = {
  agGridReact: {
    props: {
      calendarProps: {
        ...eventCalendarTestProps,
        classes: {
          defaultCheckbox: 'defaultCheckbox',
          loadingErrorIcon: 'loading-error',
          noCustomTags: 'no-custom-tags',
        },
        setExpandedStyle: jest.fn(),
      },
    },
  },
  api: {
    redrawRows: jest.fn(),
  },
  eventsCalendarSettings,
  node: {
    data: {},
    expanded: false,
    field: 'filterType',
    group: true,
    key: 'sources',
    selected: true,
    setExpanded: jest.fn(),
    setSelected: jest.fn(),
  },
  showAllColors: false,
};

export const statelessSidebarDefaultTabProps = {
  ...sidebarDefaultTabProps,
  checkboxColor: 'eventManager-checked',
  checkboxTitle: 'Event Manager',
  classes: {},
  filterStructure: ['eventManager'],
  handleRefetchGroup: jest.fn(),
  isGroup: true,
  isLoading: false,
  setCalendarSettings: jest.fn(),
  toggleGroups: jest.fn(),
  toggleStatuses: jest.fn(),
};

export const sidebarEnvPlatformProjectsFiltersProps = {
  agGridReact: {
    props: {
      calendarProps: {
        ...eventCalendarTestProps,
      },
    },
  },
  api: { redrawRows: jest.fn() },
  eventsCalendarSettings,
  node: {
    data: {
      group: 'envType',
      platform: 'CROSSPLAY',
    },
    selected: true,
    setExpanded: jest.fn(),
    setSelected: jest.fn(),
  },
  toggleSelectedFilters: jest.fn(),
  toggleGroupsExpanded: jest.fn(),
  value: 'CROSSPLAY',
};

export const tagFiltersProps = {
  checked: true,
  classes: {},
  deleteCustomTagSet: jest.fn(),
  eventsCalendarSettings,
  filters: eventsCalendarSettings.filters,
  idx: '0',
  selectedStyle: 'sources',
  setCalendarSettings: jest.fn(),
  setTagsExpanded: jest.fn(),
  tags: ['COD', 'PS4', 'XB1'],
  toggleCustomTags: jest.fn(),
  value: 'CROSSPLAY',
};

export const datePickerFieldsProps = {
  eventsCalendarSettings: {
    ...eventsCalendarSettings,
    datePickerProps: {
      disableDateSelection: jest.fn(),
      project: {
        id: 1,
        name: 'Test Project',
      },
      setEndDate: jest.fn(),
      setEndValue: jest.fn(),
      setEndValueError: jest.fn(),
      setStartDate: jest.fn(),
      setStartValue: jest.fn(),
      setStartValueError: jest.fn(),
    },
  },
  onChangeDatePickerSelection: jest.fn(),
  userTimezone: eventCalendarTestProps.userTimezone,
};

export const mockState = {
  Core: {
    EventsCalendar: eventsCalendarSettings,
  },
};

export const mockStateFn = eventsCalendarSettingsOverride => ({
  Core: {
    EventsCalendar: eventsCalendarSettingsOverride,
  },
});
export const eventFilterProps = {
  eventGroups,
  eventsCalendarSettings,
  disabledFilters: {},
  onFetchEvents: jest.fn(),
};

export const eventFilterGroupProps = {
  group: {
    disabled: true,
    groupProps: { classes: {}, loading: { error: null, isLoading: false } },
    isGroup: true,
    items: [
      {
        isSubgroup: true,
        items: [
          { name: 'active', items: false, isSubgroup: false },
          { name: 'analysis', items: false, isSubgroup: false },
          { name: 'archived', items: false, isSubgroup: false },
          { name: 'config', items: false, isSubgroup: false },
          { name: 'killed', items: false, isSubgroup: false },
        ],
        name: 'demonwareABTesting',
      },
    ],
    name: 'abTesting',
  },
  eventGroups,
  eventsCalendarSettings,
  onFetchEvents: jest.fn(),
  setCalendarSettings: jest.fn(),
};
