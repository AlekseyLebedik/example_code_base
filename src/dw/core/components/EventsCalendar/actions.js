import * as AT from './actionTypes';

export const setEventsCalendarSettings = calendarSettings => ({
  type: AT.SET_EVENTS_CALENDAR_SETTINGS,
  calendarSettings,
});

export const clearEventsCalendarSettings = () => ({
  type: AT.CLEAR_EVENTS_CALENDAR_SETTINGS,
});
