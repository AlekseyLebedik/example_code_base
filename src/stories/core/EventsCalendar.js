/* eslint-disable import/no-extraneous-dependencies */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cloneDeep from 'lodash/cloneDeep';
import { createStore, combineReducers } from 'redux';

import { withInfo } from '@storybook/addon-info';
import { storiesOf } from '@storybook/react';

import reduxDecorator from 'stories/helpers/reduxDecorator';

import EventsCalendar from 'dw/core/components/EventsCalendar';
import eventsCalendarReducer from 'dw/core/components/EventsCalendar/reducer';

import * as CoreReplicas from 'dw/core/replicas';

import { CreateEventManagerEventModal } from './EventsCalendar/createEventManagerEvent';
import { eventsCalendarIntro } from './EventsCalendar/intro';
import { otherTestEventGroup } from './EventsCalendar/otherGroupData';
import { eventManagerTestEventGroup } from './EventsCalendar/eventManagerGroupData';

import styles from './EventsCalendar/eventsCalendar.module.css';

const { UserReplica } = CoreReplicas;

const calendarPlatforms = ['CROSSPLAY', 'PC', 'PS4', 'XB1'];

const rootReducer = combineReducers({
  Core: combineReducers({
    EventsCalendar: eventsCalendarReducer,
  }),
  user: UserReplica.reducer,
});

const store = createStore(rootReducer);

class WrappedCalendar extends Component {
  static propTypes = {
    eventGroups: PropTypes.arrayOf(PropTypes.object),
    SelectSlotComponent: PropTypes.elementType,
  };

  static defaultProps = {
    eventGroups: [],
    SelectSlotComponent: undefined,
  };

  constructor(props) {
    super(props);

    this.state = {
      eventGroups: props.eventGroups.map(group => ({
        ...group,
        modifyEvent: this.editEvent,
      })),
      isCreateEventVisible: false,
      projects: [
        { projectId: 1, name: 'GTR Project' },
        { projectId: 'null', name: 'Cross Project' },
      ],
    };
  }

  closeModal = () => {
    this.setState({
      isCreateEventVisible: false,
    });
  };

  openModal = () => {
    this.setState({
      isCreateEventVisible: true,
    });
  };

  editEvent = event => {
    this.setState(state => {
      let updatedEventGroups = cloneDeep(state.eventGroups);
      updatedEventGroups = updatedEventGroups.map(group => {
        if (group.type === event.type) {
          const eventIdx = group.events.findIndex(e => e.id === event.id);
          const updatedEvents = cloneDeep(group.events);
          updatedEvents[eventIdx] = event;
          return {
            ...group,
            events: updatedEvents,
          };
        }
        return group;
      });

      return { eventGroups: updatedEventGroups };
    });
  };

  render() {
    const { eventGroups, isCreateEventVisible, projects } = this.state;
    const { SelectSlotComponent } = this.props;

    return (
      <div className={styles.calendarStoryWrapper}>
        {isCreateEventVisible && SelectSlotComponent && (
          <SelectSlotComponent
            closeModal={this.closeModal}
            isCreateEventVisible={isCreateEventVisible}
          />
        )}
        <EventsCalendar
          {...this.props}
          affiliatedProjects={projects}
          eventGroups={eventGroups}
          onFetchEvents={() => {}}
          onSelectSlot={SelectSlotComponent && this.openModal}
          platforms={calendarPlatforms}
        />
      </div>
    );
  }
}

storiesOf('core/EventsCalendar', module)
  .addDecorator(reduxDecorator(store))
  .addDecorator(withInfo())
  .add('intro', eventsCalendarIntro)
  .add('no sidebar', () => <WrappedCalendar eventGroups={[]} sidebar={false} />)
  .add('no event groups', () => <WrappedCalendar eventGroups={[]} sidebar />)
  .add('one event group (with create event)', () => (
    <WrappedCalendar
      eventGroups={[eventManagerTestEventGroup]}
      SelectSlotComponent={CreateEventManagerEventModal}
      sidebar
    />
  ))
  .add('two event groups', () => (
    <WrappedCalendar
      eventGroups={[eventManagerTestEventGroup, otherTestEventGroup]}
      sidebar
    />
  ));
