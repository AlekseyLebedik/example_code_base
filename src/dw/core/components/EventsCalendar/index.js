import * as actions from './actions';
import * as selectors from './selectors';
import * as helpers from './helpers';
import * as sidebarHooks from './components/EventsCalendarSidebar/hooks';
import * as hooks from './hooks';
import * as constants from './constants';
import * as themeEventsFilter from './themeEventsFilter';

export { default } from './container';

export { default as reducer } from './reducer';

const allHooks = { ...sidebarHooks, ...hooks };

export {
  actions,
  selectors,
  helpers,
  allHooks as hooks,
  themeEventsFilter,
  constants,
};
