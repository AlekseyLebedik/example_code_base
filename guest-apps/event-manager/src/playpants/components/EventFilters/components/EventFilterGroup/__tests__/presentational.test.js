import React from 'react';
import { shallow } from 'enzyme';

import { eventCalendarTestProps as props } from 'dw/core/components/EventsCalendar/testData';

import EventFilterGroupStateless from '../presentational';

describe('EventFilterGroupStateless', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<EventFilterGroupStateless {...props} />);
  });

  it('renders default EventFilterGroupStateless empty', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
