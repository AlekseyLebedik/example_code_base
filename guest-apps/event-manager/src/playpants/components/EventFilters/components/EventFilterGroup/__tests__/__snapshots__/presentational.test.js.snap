// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventFilterGroupStateless renders default EventFilterGroupStateless empty 1`] = `
<WithStyles(ForwardRef(List))
  className="list"
>
  <div
    className="listSubheader"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      checked={false}
      classes={
        Object {
          "label": undefined,
          "root": undefined,
        }
      }
      control={
        <WithStyles(ForwardRef(Tooltip))
          disableFocusListener={true}
          disableHoverListener={true}
          disableTouchListener={true}
          title="Click to retry undefined source fetch"
        >
          <WithStyles(ForwardRef(Checkbox))
            classes={
              Object {
                "checked": undefined,
                "root": "checkbox",
              }
            }
            disableRipple={true}
            indeterminate={false}
            onChange={[Function]}
          />
        </WithStyles(ForwardRef(Tooltip))>
      }
      key="calendarEventSelector/undefined"
    />
    <WithStyles(ForwardRef(Tooltip))
      placement="right"
      title="Color by Event undefined"
    >
      <WithStyles(ForwardRef(Radio))
        checked={false}
        className="radioButton"
        color="primary"
        disableRipple={true}
        onClick={[Function]}
        size="small"
      />
    </WithStyles(ForwardRef(Tooltip))>
  </div>
</WithStyles(ForwardRef(List))>
`;
